components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/ValidationImpl.java

```diff
@@ -80,135 +70,58 @@
-            response.errorType = "existingPedigree";
-            response.message =
-                String.format("patient %s already has a different pedigree, and therefore cannot be included in "
-                    + "this one.", patientId);
-            return response;
-        }
+        User currentUser = this.userManager.getCurrentUser();
+        return hasPatientAccess(patient, this.editAccess, currentUser);
     }
 
     @Override
-    public StatusResponse canAddEveryMember(XWikiDocument family, List<String> updatedMembers)
-        throws XWikiException
-    {
-        StatusResponse defaultResponse = new StatusResponse();
-        defaultResponse.statusCode = 200;
-
-        for (String member : updatedMembers) {
-            StatusResponse patientResponse = this.canAddToFamily(family, member);
-            if (patientResponse.statusCode != 200) {
-                return patientResponse;
-            }
-        }
-        return defaultResponse;
-    }
-
-    private boolean safeIsInFamilyCheck(XWikiDocument familyDoc, String patientId) throws XWikiException
+    public boolean hasPatientEditAccess(Patient patient, User user)
     {
-        if (familyDoc != null) {
-            return this.isInFamily(familyDoc, patientId);
-        }
-        return false;
+        return hasPatientAccess(patient, this.editAccess, user);
     }
 
     @Override
-    public StatusResponse checkFamilyAccessWithResponse(XWikiDocument familyDoc)
+    public boolean hasPatientViewAccess(Patient patient)
     {
-        StatusResponse response = new StatusResponse();
         User currentUser = this.userManager.getCurrentUser();
-        if (this.authorizationService.hasAccess(currentUser, Right.EDIT,
-            new DocumentReference(familyDoc.getDocumentReference())))
-        {
-            response.statusCode = 200;
-            return response;
-        }
-        response.statusCode = 401;
-        response.errorType = "permissions";
-        response.message = "Insufficient permissions to edit the family record.";
-        return response;
+        return hasPatientAccess(patient, this.viewAccess, currentUser);
     }
 
-    /* Should not be used when saving families. Todo why? */
     @Override
-    public boolean hasPatientEditAccess(String patientId)
+    public boolean hasPatientViewAccess(Patient patient, User user)
     {
-        User currentUser = this.userManager.getCurrentUser();
-        return hasPatientAccess(this.patientRepository.getPatientById(patientId), this.editAccess, currentUser);
+        return hasPatientAccess(patient, this.viewAccess, user);
     }
 
-    @Override
-    public boolean hasPatientEditAccess(Patient patient, User user)
+    private boolean hasPatientAccess(Patient patient, AccessLevel accessLevel, User user)
     {
-        return hasPatientAccess(patient, this.editAccess, user);
+        PatientAccess patientAccess = this.permissionsManager.getPatientAccess(patient);
```

Comment:
One thing I don't like is that family studies now makes the inclusion of `patient-access-rules` mandatory. It should be kept optional, so only the `authorizationService` should be used, not the `permissionsManager`. Is that possible?